.github/workflows/build.yml

```diff
@@ -7,19 +7,23 @@ on:
     - 'master'
     tags:
     - 'source/v*.*.*'
+  pull_request:
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
 
-    - name: Set-up JDK 11
-      uses: actions/setup-java@v2
+    - name: Validate wrapper
+      uses: gradle/wrapper-validation-action@v1
+
+    - name: Set-up JDK 17
+      uses: actions/setup-java@v3
       with:
-        distribution: 'temurin'
-        java-version: '11'
+        distribution: 'zulu'
+        java-version: '17'
```

Comment:
According to [WhichJDK](https://whichjdk.com), [Adoptium Temurin](https://adoptium.net) should be the best choice. Why zulu?